app/models/project_media.rb

```diff
@@ -475,21 +475,95 @@ def get_similar_articles
       search_query ||= self.title
       results = self.team.search_for_similar_articles(search_query, self)
       fact_check_ids = results.select{|article| article.is_a?(FactCheck)}.map(&:id)
+      fc_pm = {}
+      unless fact_check_ids.blank?
+        # Get ProjectMedia for FactCheck for RelevantResultsItem logs and should use sort_by to keep existing order
+        FactCheck.select('fact_checks.id AS fc_id, claim_descriptions.project_media_id AS pm_id')
+        .where(id: fact_check_ids).joins(:claim_description).each do |raw|
+          fc_pm[raw.fc_id] = raw.pm_id
+        end
+      end
       explainer_ids = results.select{|article| article.is_a?(Explainer)}.map(&:id)
-      { fact_check: fact_check_ids, explainer: explainer_ids }.to_json
+      ex_pm = {}
+      unless explainer_ids.blank?
+        # Intiate the ex_pm with nil values as some Explainer not assinged to existing items
+        default_pm = nil
+        ex_pm = explainer_ids.each_with_object(default_pm).to_h
+        # Get ProjectMedia for Explainer for RelevantResultsItem logs and should use sort_by to keep existing order
+        ExplainerItem.where(explainer_id: explainer_ids).find_each do |raw|
+          ex_pm[raw.explainer_id] = raw.project_media_id
+        end
+      end
+      threshold = Bot::Alegre.get_threshold_for_query('text', self)[0][:value]
+      {
+        fact_check: fc_pm.sort_by { |k, _v| fact_check_ids.index(k) }.to_h,
+        explainer: ex_pm.sort_by { |k, _v| explainer_ids.index(k) }.to_h,
+        similarity_settings: { threshold: threshold }
+      }.to_json
     end
     if results.blank?
       # This indicates a cache hit, so we should retrieve the items according to the cached values while maintaining the same sort order.
       items = JSON.parse(items)
-      items.each do |klass, ids|
+      items.each do |klass, data|
+        ids = data.keys
         results += klass.camelize.constantize.where(id: ids).sort_by { |result| ids.index(result.id) }
       end
     end
     results
   end
 
+  def log_relevant_results(article, author_id)
+    data = begin JSON.parse(Rails.cache.read("relevant-items-#{self.id}")) rescue {} end
+    type = article.class.name.underscore
+    unless data[type].blank?
+      user_action = data[type].include?(article.id) ? 'relevant_articles' : 'article_search'
+      similarity_settings = data['similarity_settings']
+      items = data[type]
+      items.keys.each_with_index do |value, index|
+        selected_count = (value == article.id).to_i
+        fields = {
+          article_id: article.id,
+          article_type: article.class.name,
+          matched_media_id: items[value],
+          selected_count: selected_count,
+          display_rank: index + 1,
+        }
+        create_relevant_results_item(user_action, similarity_settings, author_id, fields)
```

Comment:
Can you please call it with `self` here? Just to avoid confusion with the `SampleData` method of the same name.

Reply:
Done